Flaky unit test: TestStagedFileWriter_Close_ErrorCleansUpResources (stagedstorage)

## Summary

`TestStagedFileWriter_Close_ErrorCleansUpResources` in `server/storage/stagedstorage` is **flaky**. It fails intermittently in CI (observed on the PR #173 "Unit Test" job) with:

```
--- FAIL: TestStagedFileWriter_Close_ErrorCleansUpResources (0.10s)
    writer_impl_test.go:2573:
        Error:   An error is expected but got nil.
        Messages: Close should return error when awaitAllFlushTasks fails
FAIL  github.com/zilliztech/woodpecker/server/storage/stagedstorage
```

## Root cause (test-only; production code is correct)

The test forces `awaitAllFlushTasks` to fail by cancelling `runCtx` + passing an already-cancelled `ctx` to `Close`, then asserts `Close` returns an error.

But `awaitAllFlushTasks` (`writer_impl.go`) uses a `select` with three simultaneously-ready branches:

```go
select {
case f.flushTaskChan <- &blockFlushTask{entries: nil, ...}: // buffered chan (cap >= 300) -> always ready
    ...
case <-ctx.Done():      return ctx.Err()
case <-f.runCtx.Done(): return werr.ErrFileWriterAlreadyClosed
}
```

Go picks a ready case at random. When it picks the **send** branch, the flush-queue processor consumes that termination signal and sets `allUploadingTaskDone = true`; the subsequent loop then hits `if allUploadingTaskDone { return nil }`, so `awaitAllFlushTasks` returns `nil` and `Close` returns `nil` → the assertion fails.

The production `Close()` error-propagation and resource-cleanup logic is correct — only the test's failure-injection mechanism is non-deterministic.

## Reproduction

```bash
go test ./server/storage/stagedstorage/ \
  -run '^TestStagedFileWriter_Close_ErrorCleansUpResources$' -count=200 -failfast
```

Fails within ~200 iterations. The failing run logs `termination signal sent to ack goroutine` followed by `shared flush worker completed processing all tasks` (instead of `wait flush error before close`).

## Proposed fix

Test-only. Block the flush-queue processor from starting during `Close` (set `flushTasksQueueProcessing = true`) so the termination signal can never be consumed and `allUploadingTaskDone` cannot flip to `true` mid-close — leaving the cancelled `ctx`/`runCtx` as the only exits, both of which return an error. Deterministic.

Verified: `-count=2000` green, `-race -count=50` green, full package green.

--- a/server/storage/stagedstorage/writer_impl_test.go
+++ b/server/storage/stagedstorage/writer_impl_test.go
@@ -2566,6 +2566,14 @@ func TestStagedFileWriter_Close_ErrorCleansUpResources(t *testing.T) {
 	writer.runCancel()
 	time.Sleep(100 * time.Millisecond)
 
+	// Block the flush-queue processor from (re)starting during Close. Otherwise
+	// awaitAllFlushTasks can win the race where its select takes the buffered
+	// "send termination signal" branch, a processor consumes that signal and
+	// flips allUploadingTaskDone to true, and awaitAllFlushTasks returns nil --
+	// making this test flaky. With no processor to consume the signal, the
+	// cancelled ctx/runCtx deterministically drive awaitAllFlushTasks to error.
+	writer.flushTasksQueueProcessing.Store(true)
+
 	// Close with an already-cancelled context so awaitAllFlushTasks returns error immediately
 	cancelledCtx, cancel := context.WithCancel(context.Background())
 	cancel()

